src/util/fileIOhooks.ts

```diff
@@ -0,0 +1,85 @@
+import { useFileDialog } from '@mantine/hooks';
+
+function useLoadJsonFile(handlePickedFile: (content: JSON) => void): () => void {
+  const fileDialog = useFileDialog({
+    multiple: false,
+    accept: '.json',
+    resetOnOpen: true,
+    onChange: readFile
+  });
+
+  async function readFile(contents: FileList | null) {
+    try {
+      const content = await contents?.[0]?.text();
+      if (!content) {
+        return;
+      }
+      const json = JSON.parse(content);
+      handlePickedFile(json);
+    } catch (e) {
+      // TODO: do we want to throw here?
+      console.error('Error parsing file', e);
+    }
+  }
+
+  function openLoadFileDialog() {
+    fileDialog.open();
+  }
+
+  return openLoadFileDialog;
+}
+
+// For documentation about these features please read this article:
+// https://developer.chrome.com/docs/capabilities/browser-fs-access#opening_files_2
+async function openSaveFileDialog(contents: JSON) {
+  const contentsString = JSON.stringify(contents, null, 2);
+
+  const supportsSaveDialog = 'showSaveFilePicker' in self;
+  if (supportsSaveDialog) {
+    const options = {
+      types: [
+        {
+          description: 'Text Files',
+          accept: {
+            'text/plain': ['.json']
+          }
+        }
+      ]
+    };
+    // For some reason typescript doesn't recognize this as a function
+    // It is an experimental feature of the chromium browser
+    const fileHandle = await window.showSaveFilePicker(options);
```

Comment:
This needs to be wrapped in a try/catch if the users cancels the window dialog an error is thrown. Don't think we need to handle it in anyway just catch it and continue with our day